### DISCUSSION TITLE
issue when upgrade from 1.6 to 2.2

### DISCUSSION THREAD

**Discussion by JayHJ** [MorphiaOrg/morphia#2105] [General]
** Please ask your question and provide whatever background or supporting information you think might be relevant: **
we have a complex data structure which defined in the below format, and when we upgrade from 1.6 to 2.2.3, we got issue that the order by throw error. can you please help on this?

Entities structure:
```java
@Entity
class A {
    @Id
    private ObjectId id;
    private B event;
}

@Entity
interface B {
}

@Entity
class C implements B {
    setEventData()
}

@Entity
class D extends E {
}

@Entity
abstract class E extends F {
    private long startTime;
}

abstract class F implements G {
}
```

now we build the object in below way:

```java
A data = new A();
D data1 = new D();
data1.setStartTime(2222222L);

C data2 = new C();
data2.setEventData(data1);

data.setEvent(data2)
```

so in this way, we have path: event.eventData.startTime, but when we try to use order by this path, it show error unknown path

** Please complete the following information: **
    Server Version:
    Driver Version: 4.4.2
    Morphia Version: 2.2.3

**Comment by evanchooly** [comment]
My apologies for the delay.  I've been traveling for working and trying to get caught up on everything. I started to try to recreate this from what you've shown here but there are enough gaps that I'm having to fill in that I'm losing confidence that I properly recreating your issue.  If you could put together a [reproducer](/morphiaorg/reproducer) I'd be happy to take a look and see.

**Reply by JayHJ** [reply]
[reproducer-master.zip](https://github.com/MorphiaOrg/morphia/files/9791139/reproducer-master.zip)

thank you for help. please find the reproducer attached.

**Comment by JayHJ** [comment]
hi @evanchooly , any suggestion for this issue? thank you.

**Comment by evanchooly** [comment]
My apologies.  I'm in a bit of a frenzy trying to wrap up 2.3.  But the fix to your problem is simple: you need to call Mapper.map() on your entities.  You can either list them all individually or you can map a whole package at once with `mapPackage()'.  Give that a shot and see how it goes.  Let me know if you have any further questions.
